### PR TITLE
fix(connector-insights): ignore CI report if it does not exist

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

Fixes a bug in connector insights not generating for a bunch of connectors when the CI report for their test run does not exist: https://github.com/airbytehq/airbyte/actions/workflows/connectors_insights.yml

Not that we assumed those JSON reports are there, but for some reason they are not. I'm not sure if it's a bug and the report path assumed previous version of the connector, or if we stopped putting the reports there.

## How

Wrapped accessing the report in a try-catch block. This is a band-aid, and a better fix would be preferred.

@alafanechere would have to remove SBOM generation from connector-insights anyway, so perhaps it's worth making a better fix? 

## User Impact
None.

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by ChristoGrab — APPROVED** [review]
Seal of approval granted

image

**Comment by ChristoGrab** [comment]
Approved to unblock, but switched off automerge in case Augustin does have a suggestion for a more preferable fix!

**Review by alafanechere — APPROVED** [review]
I think I know the root cause of the absence of reports:
I've recently set a 45 days retention on reports in GCS.
Community connectors (which are not nightly tested) will not get fresh nightly reports.

File: airbyte-ci/connectors/connectors_insights/src/connectors_insights/hacks.py [hunk 1]
```diff
@@ -56,10 +56,13 @@ def get_ci_on_master_report(connector: Connector) -> Dict | None:
...
-            # The first table entry is the latest report
-            json_report = get_ci_json_report(json_report_url)
-            if json_report["connector_version"] == connector.version:
-                return json_report
+            # The first table entry is the latest report, but sometimes it's not there questionmark
+            try:
+                json_report = get_ci_json_report(json_report_url)
+                if json_report["connector_version"] == connector.version:
+                    return json_report
+            except Exception as e:
```
Comment (alafanechere) [inline airbyte-ci/connectors/connectors_insights/src/connectors_insights/hacks.py:64]: I'd recommend catching a less broad exception.

Comment (natikgadzhi) [reply airbyte-ci/connectors/connectors_insights/src/connectors_insights/hacks.py:64]: I'll merge this to get this going for now, but will return to this

**Pull request merged**